packages/flutter_tools/lib/src/base/context.dart

```diff
@@ -3,78 +3,39 @@
 // found in the LICENSE file.
 
 import 'dart:async';
+import 'package:stack_trace/stack_trace.dart';
 
 final AppContext _defaultContext = new AppContext();
 
 typedef void ErrorHandler(dynamic error, StackTrace stackTrace);
 
-/// A singleton for application functionality. This singleton can be different
-/// on a per-Zone basis.
-AppContext get context {
```

Comment:
If we expect to have multiple contexts alive at the same time than the current design of keeping it inside the zone makes sense to me. I'm just trying to remove unnecessary complexity here. 

If we want to support multiple contexts simultaneously, I would like the code base to force everyone to enter an AppContext zone before running any code. The code today (ignoring this PR) falls back to the default context in the case that there is no current zone.